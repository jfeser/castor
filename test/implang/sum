Select([sum(f) as s, count() as c],
  AList(r1, ATuple([AScalar(0.f), AScalar((0.g - 0.f) as d)], cross)))