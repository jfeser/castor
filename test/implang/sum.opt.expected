// Locals:
// i18 : int (persists=true)
// count16 : int (persists=false)
// cstart21 : int (persists=true)
// found_tup14 : bool (persists=false)
// cstart17 : int (persists=true)
// tup13 : tuple[int, int] (persists=false)
// sum15 : int (persists=false)
// count19 : int (persists=true)
// cstart20 : int (persists=true)
fun printer () : void {
    found_tup14 = false;
    sum15 = 0;
    count16 = 0;
    cstart17 = 0;
    i18 = 0;
    count19 = 5;
    loop (i18 < count19) {
        cstart20 = cstart17;
        cstart21 = cstart20 + 1;
        tup13 = (buf[cstart20 : 1], buf[cstart21 : 1]);
        sum15 = sum15 + tup13[0];
        count16 = count16 + 1;
        found_tup14 = true;
        cstart17 = cstart17 + 2;
        i18 = i18 + 1;
    }
    if (found_tup14) {
        print(tuple[int, int], (sum15, count16));
    } else {
         
    }
}
// Locals:
// count8 : int (persists=true)
// i7 : int (persists=true)
// cstart9 : int (persists=true)
// sum4 : int (persists=false)
// count5 : int (persists=false)
// tup2 : tuple[int, int] (persists=false)
// found_tup3 : bool (persists=false)
// cstart10 : int (persists=true)
// cstart6 : int (persists=true)
fun consumer () : void {
    found_tup3 = false;
    sum4 = 0;
    count5 = 0;
    cstart6 = 0;
    i7 = 0;
    count8 = 5;
    loop (i7 < count8) {
        cstart9 = cstart6;
        cstart10 = cstart9 + 1;
        tup2 = (buf[cstart9 : 1], buf[cstart10 : 1]);
        sum4 = sum4 + tup2[0];
        count5 = count5 + 1;
        found_tup3 = true;
        cstart6 = cstart6 + 2;
        i7 = i7 + 1;
    }
    if (found_tup3) {
        consume(tuple[int, int], (sum4, count5));
    } else {
         
    }
}
